src/game/propobj.c

```diff
@@ -18471,7 +18472,12 @@ struct autogunobj *laptopDeploy(s32 modelnum, struct gset *gset, struct chrdata
 	if (index >= 0 && index < g_MaxThrownLaptops) {
 		setupLoadModeldef(modelnum);
 		modeldef = g_ModelStates[modelnum].modeldef;
-		laptop = &g_ThrownLaptops[index];
+		s32 laptopIndex = index * g_MaxThrownLaptopsPerPlayer;
```

Comment:
This is when the user sets up a new sentry. All sentries are stored in one big array, so now that each player can have multiple, let's multiply the player index with the number of sentries per player to find the start of the searching space. Then let's iterate over the space allocated for that user until we find a free space to store that new sentry gun in memory.